--- mtd-utils-1.3.1/ubi-utils/old-utils/src/error.c.bak	2011-11-02 10:42:39.640872202 +0800
+++ mtd-utils-1.3.1/ubi-utils/old-utils/src/error.c	2011-11-02 10:43:22.360875043 +0800
@@ -74,13 +74,11 @@
 	FILE* fpout;
 	char buf[MAXLINE + 1];
 	va_list	ap;
-	int n;
 
 	fpout = stdout;
 
 	va_start(ap, fmt);
 	vsnprintf(buf, MAXLINE, fmt, ap);
-	n = strlen(buf);
 	strcat(buf, "\n");
 
 	fputs(buf, fpout);
--- mtd-utils-1.3.1/ubi-utils/old-utils/src/libubigen.c.bak	2011-11-02 16:14:32.955539370 +0800
+++ mtd-utils-1.3.1/ubi-utils/old-utils/src/libubigen.c	2011-11-02 16:14:48.292186360 +0800
@@ -201,7 +201,7 @@
 	u->v->lnum = cpu_to_be32(u->blks_written);
 
 	if (action & MARK_AS_UPDATE) {
-		u->v->copy_flag = (u->v->copy_flag)++;
+		(u->v->copy_flag)++;
 	}
 
 	write_vid_hdr(u, action);
